apiserver/facades/client/application/application.go

```diff
@@ -2531,6 +2540,29 @@ func (api *APIBase) MergeBindings(ctx context.Context, in params.ApplicationMerg
 	return params.ErrorResults{Results: res}, nil
 }
 
+// convertSpacesToIDInBindings takes the input bindings (which contain space
+// names) and converts them to spaceIDs.
+// TODO(nvinuesa): this method should not be needed once we migrate endpoint
+// bindings to dqlite.
+func (api *APIBase) convertSpacesToIDInBindings(bindings map[string]string) (map[string]string, error) {
```

Comment:
You're identifying places where you'll need your service in order to drop the spaces from Mongo.